### DISCUSSION TITLE
1 time compile / error / short hint

### DISCUSSION THREAD

**Discussion by Mr-Rog** [Jyers/Marlin#198] [Q&A] [answered]
Hi Jyers cool Firmware Community
I try tol learn, how i can complime, so when I try this the first time with nothing change 
[error.txt](https://github.com/Jyers/Marlin/files/6096604/error.txt)
I get this Error
as written, I want to learn the basics, also continue to watch videos.
Thanks for your help
Stay healthy
Rog

Edit I aee this
#if ENABLED(Z_MIN_PROBE_REPEATABILITY_TEST)
    #error "Z_MIN_PROBE_REPEATABILITY_TEST requires a probe: FIX_MOUNTED_PROBE, NOZZLE_AS_PROBE, BLTOUCH, SOLENOID_PROBE, Z_PROBE_ALLEN_KEY, Z_PROBE_SLED, or Z Servo."
  #endif

**Comment by Jyers** [comment, marked as answer]
It would seem you have the Z_MIN_PROBE_REPEATABILITY_TEST enabled for some reason. Just find the line "#define Z_MIN_PROBE_REPEATABILITY_TEST" and change it to "//#define Z_MIN_PROBE_REPEATABILITY_TEST"

**Comment by Mr-Rog** [comment]
Hello  Jacob
thank you for your hint, i had nothing change, but maybe I use the wrong examples, 
Short question, is it possible to get the template for (E3V2-BLTouch-5x5-HS-V4.bin) so that I can change my 4 settings to my Printer-Config
Configuration Files in this folder im not sure which config are for the E3V2-BLTouch-5x5-HS-V4.
I would also very much like to make a donation, it's not like you have the big work and I just want to profit from it.
thx for a short Feeback
Rog

**Reply by Jyers** [reply]
To recreate the BLTouch 5x5 HS, you want to start with the BLTouch-3x3-HighSpeed Config templates and change:
"#define GRID_MAX_POINTS_X 3" -> "#define GRID_MAX_POINTS_X 5"
From there you should be good to change any settings you need for your printer and compile.
Thank you so much for supporting the development, there is a sponsor button on the top bar of the GitHub repository if you would like to make a donation.

**Comment by Mr-Rog** [comment]
cool thank you
last question
so ther is not the same \Configuration Files\Jyers's Config your config and this
https://github.com/Jyers/Marlin/releases/tag/v1.1.1?fbclid=IwAR3m1jO8h0HAf6d-VjWPWIudsxr820I-hPZjKLFOxaNr1YgJamoBiTWdICE
E3V2-BLTouch-5x5-HS-V4.
that confuses me, which file I must use when I will change settings from this E3V2-BLTouch-5x5-HS-V4, is more or less perfect smal changes.
you see my point
regads
Rog

**Reply by Jyers** [reply]
The templates are found in Configuration Files/E3V2 Templates. This is where you can find the BLTouch-3x3-HighSpeed that you will need to modify and use to compile your own .bin

**Comment by Mr-Rog** [comment]
try my best but I get this error
 static_assert(nullptr == strstr(FILAMENT_RUNOUT_SCRIPT, "M600"), "ADVANCED_PAUSE_FEATURE is required to use M600 with FILAMENT_RUNOUT_SENSOR.");
  #endif
n file included from Marlin\src\HAL\STM32F1\tft\../../../inc/MarlinConfig.h:49:0,
                 from Marlin\src\HAL\STM32F1\tft\tft_fsmc.cpp:23:
Marlin\src\HAL\STM32F1\tft\../../../inc/SanityCheck.h:853:5: error: static assertion failed: ADVANCED_PAUSE_FEATURE is required to use M600 with FILAMENT_RUNOUT


I add
#define FIL_RUNOUT_ENABLED_DEFAULT true

but I get 2 Problems
with static_assert(nullptr 
{
	"resource":"/g:/Marlin-own/Marlin-Ender_3_V2_Extensible_UI (1)/Marlin-Ender_3_V2_Extensible_UI/Marlin/src/inc/SanityCheck.h",
	}, "owner": "C/C++",
	}, "code": "28",
	"severity": 8,
	"message": "The expression must have a constant value. --"\"function \"strstr\" (declared on line 48 of \"C:\Users\\xxxx\\.platformio\packages\\toolchain-gccarmnoneeabi\arm-none-eabi\include\string.h\")\" cannot be called if the type is not \"constexpr\".
	"source": "C/C++",
	"startLineNumber": 853,
	"startColumn": 19,
	"endLineNumber": 853,
	"endColumn": 26
}

Translated with www.DeepL.com/Translator (free version)

**Reply by Jyers** [reply]
It would seem you might not be using both of the correct files, make sure you copy both the configuration.h and the configuration_adv.h from the BLTouch-3x3-HighSpeed folder to the marlin folder If that still doesn't work you can manually fix it by enabling ADVANCED_PAUSE_FEATURE in configuration_adv.h

**Reply by Mr-Rog** [reply]
im beginn from scratch and now was working, thx for your help
